src/main/java/org/openmrs/maven/plugins/ModuleInstall.java

```diff
@@ -114,20 +116,67 @@ public void installModule(String serverId, String groupId, String artifactId, St
             if (currentProperties != null) serverId = currentProperties.getName();
         }
         File serverPath = helper.getServerPath(serverId);
-        Artifact artifact = getArtifactForSelectedParameters(helper, groupId, artifactId, version);
-        String originalId = artifact.getArtifactId();
-        artifact.setArtifactId(artifact.getArtifactId() + "-omod");
-        File modules = new File(serverPath, SDKConstants.OPENMRS_SERVER_MODULES);
-        modules.mkdirs();
-        Element[] artifactItems = new Element[1];
-        artifactItems[0] = artifact.toElement(modules.getPath());
-        File[] listOfModules = modules.listFiles();
-        boolean versionUpdated = false;
-        boolean removed = false;
+        ServerConfig properties = ServerConfig.loadServerConfig(serverPath);
+        Artifact artifact = checkOpenmrsCoreVersion(helper, properties);
+        List<Element> artifactItems = new ArrayList<Element>();
+        if(artifact != null){
+            artifactItems.add(artifact.toElement(serverPath.toString()));
+
+            executeMojoPlugin(artifactItems);
+
+            File openmrsCorePath = new File(serverPath, "openmrs-" + properties.getOpenmrsCoreVersion() + ".war");
+            openmrsCorePath.delete();
+
+            properties.setParam("openmrs.platform.version", mavenProject.getVersion());
+            properties.save();
```

Comment:
I'd put lines 119-131 in a separate method: boolean upgradeOpenmrsWar(...), which returns true if it was upgraded